utils/compare.sh

```diff
@@ -28,44 +28,49 @@ check_metric_to_modify() {
 }
 
 run_benchmark_comparison() {
-   log "benchmark"
-   if [[ -n ${ES_SERVER} ]] && [[ -n ${COMPARISON_CONFIG} ]]; then
-    
-     log "Installing touchstone"
-     install_touchstone
-     network_ns=openshift-ovn-kubernetes
-     get_network_type
-     export TOUCHSTONE_NAMESPACE=${TOUCHSTONE_NAMESPACE:-"$network_ns"}
-     res_output_dir="/tmp/${WORKLOAD}-${UUID}"
-     mkdir -p ${res_output_dir}
-     export COMPARISON_OUTPUT=${PWD}/${WORKLOAD}-${UUID}.csv
-     final_csv=${res_output_dir}/${UUID}.csv
-     echo "final csv $final_csv"
-     for config in ${COMPARISON_CONFIG}
-     do
-       config_loc=$(dirname $(realpath ${BASH_SOURCE[0]}))/touchstone-configs/${config}
-       echo "config ${config_loc}"
-       check_metric_to_modify $config_loc
-       COMPARISON_FILE="${res_output_dir}/${config}"
-       envsubst < $config_loc > $COMPARISON_FILE
-       echo "comparison output"
-       if [[ -n ${ES_SERVER_BASELINE} ]] && [[ -n ${BASELINE_UUID} ]]; then
-         log "Comparing with baseline"
-         compare "${ES_SERVER_BASELINE} ${ES_SERVER}" "${BASELINE_UUID} ${UUID}" "${COMPARISON_FILE}" "${GEN_CSV}"
-       else
-         log "Querying results"
-         compare ${ES_SERVER} ${UUID} "${COMPARISON_FILE}" "${GEN_CSV}"
-       fi
-        log "python csv modifier"
-        python $(dirname $(realpath ${BASH_SOURCE[0]}))/csv_modifier.py -c ${COMPARISON_OUTPUT} -o ${final_csv}
-     done
-     if [[ -n ${GSHEET_KEY_LOCATION} ]] && [[ ${GEN_CSV} == true ]] ; then
-       gen_spreadsheet ${WORKLOAD} ${final_csv} ${EMAIL_ID_FOR_RESULTS_SHEET} ${GSHEET_KEY_LOCATION}
-     fi
-     log "Removing touchstone"
-     remove_touchstone
+  log "benchmark"
+  compare_result=0
+  if [[ -n ${ES_SERVER} ]] && [[ -n ${COMPARISON_CONFIG} ]]; then
+  
+    log "Installing touchstone"
+    install_touchstone
+    network_ns=openshift-ovn-kubernetes
+    get_network_type
+    export TOUCHSTONE_NAMESPACE=${TOUCHSTONE_NAMESPACE:-"$network_ns"}
+    res_output_dir="/tmp/${WORKLOAD}-${UUID}"
+    mkdir -p ${res_output_dir}
+    export COMPARISON_OUTPUT=${PWD}/${WORKLOAD}-${UUID}.csv
+    final_csv=${res_output_dir}/${UUID}.csv
+    echo "final csv $final_csv"
+    for config in ${COMPARISON_CONFIG}
+    do
+      config_loc=$(dirname $(realpath ${BASH_SOURCE[0]}))/touchstone-configs/${config}
+      echo "config ${config_loc}"
+      check_metric_to_modify $config_loc
+      COMPARISON_FILE="${res_output_dir}/${config}"
+      envsubst < $config_loc > $COMPARISON_FILE
+      echo "comparison output"
+      if [[ -n ${ES_SERVER_BASELINE} ]] && [[ -n ${BASELINE_UUID} ]]; then
+        log "Comparing with baseline"
+        if ! compare "${ES_SERVER_BASELINE} ${ES_SERVER}" "${BASELINE_UUID} ${UUID}" "${COMPARISON_FILE}" "${GEN_CSV}"; then
+          compare_result=$((${compare_result} + 1))
```

Comment:
Is you intention of adding 1 to the RC to count the number of failed comparisons?

Reply:
For the router-perf-v2 change in this pr, I just cared about the return code is 0 (pass) or >0(fail), but I thought the return code to count the failed number may be useful to some other 'caller' to this function when there are multiple touchstone-configs.

Reply:
I can change it to return 1 for all failed comparison case if >1 return code are to be reserved for other error condition.

Reply:
@afcollins I updated the change to let run_benchmark_comparison function to return 1 when there is any compare failure.

Reply:
ok! I didn't have a preference. Was only clarifying :-)

I agree maybe we could use the number of failures to take some action, but I do not have one in mind at this moment.

Reply:
@afcollins I don't have too :) Would you final review and merge this?